cargo-dist/src/lib.rs

```diff
@@ -57,8 +57,90 @@ pub mod tasks;
 #[cfg(test)]
 mod tests;
 
+/// dist env test -- make sure we have everything we need for a build.
+pub fn do_env_test(cfg: &Config) -> DistResult<()> {
+    let (dist, _manifest) = tasks::gather_work(cfg)?;
+
+    let builds = dist.config.builds;
+
+    let need_cargo_auditable = builds.cargo.cargo_auditable;
+    let need_cargo_cyclonedx = builds.cargo.cargo_cyclonedx;
+    let need_omnibor = builds.omnibor;
+    let mut need_xwin = false;
+    let mut need_zigbuild = false;
+
+    let tools = dist.tools;
+
+    let cargo = tools.cargo()?;
+    let host = cargo.host_target.parse()?;
+
+    for step in dist
+        .local_build_steps
+        .iter()
+        .chain(dist.global_build_steps.iter())
```

Comment:
I don't think `BuildStep::Cargo` can ever appear in `dist.global_build_steps` - should be safe to just iterate on local build steps here.

Reply:
Good catch! At one point I wondered if this was the case, but I forgot to follow up on it.